### PR TITLE
[ioctl] Build did generate command line into new ioctl

### LINKED ISSUE #3290: [ioctl] build did generate command line into new ioctl
### What would you like to be added
 - Step 1: create `didgenerate.go` and `didgenerate_test.go` files in `./ioctl/newcmd/did`. create this directory if not exist. 
 - Step 2: refactor  command  https://github.com/iotexproject/iotex-core/blob/master/ioctl/cmd/did/didgenerate.go in `didgenerate.go`
 - Step 3: add unit test in xx_test.go

#### Notes:
1. Use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to construct the Cobra command.
2. [Output package](https://github.com/iotexproject/iotex-core/blob/master/ioctl/output) is deprecated. `e.g.` `output.PrintError` could be replaced with `errors.Wrap`.
3. Global variables (config.ReadConfig  config.DefaultConfigFile etc.) have been wrapped into client interface. Please use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to access them if needed
4. The logic of new code should be similar to [the old one](https://github.com/iotexproject/iotex-core/blob/master/ioctl/cmd/did/didgenerate.go).
5. In each unit test, the results should be checked whether they are as the expected. 

#### Example:
1. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountinfo.go
2. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountdelete.go
3. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountcreateadd.go

### Why is this needed
Unit tests couldn't be added for the old ioctl command (./ioctl/cmd)

### PR DESCRIPTION
**Pull request by LuckyPigeon**
# Description

Refactor `didgenerate` command in new `ioctl`, with the following note.

* Use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to construct the Cobra command.
* Output package is deprecated, replace it with errors package.
* Replace fmt.Println with cmd.Println
* Refactor unit test to cover the modification.

Fixes #3290 

## Type of change
Please delete options that are not relevant.
- [x] Code refactor or improvement

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [x] TestNewDidGenerateCmd

**Test Configuration**:
- Firmware version: Windows 10 (WSL Version: Ubuntu 18.04)
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by LuckyPigeon** [comment]
@huof6829 Could add a label of HACKTOBERFEST-ACCEPTED in this PR? I'm attending the annual GitHub hacktoberfest currently.

File: ioctl/newcmd/did/didgenerate.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+			if err != nil {
+				return errors.Wrap(err, "")
+			}
+
+			sum := sha256.Sum256(msg)
+			generatedMessage := string(msg) + "\n\nThe hex encoded SHA256 hash of the DID doc is:" + hex.EncodeToString(sum[:])
+			if err != nil {
+				return errors.Wrap(err, "failed to sign message")
+			}
+			cmd.Println(generatedMessage)
```
Comment (huof6829) [inline ioctl/newcmd/did/didgenerate.go:101]: delete variable generatedMessage and use cmd.Printf("  ",args...)

File: ioctl/newcmd/did/didgenerate_test.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	"github.com/golang/mock/gomock"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+
+	"github.com/iotexproject/iotex-address/address"
```
Comment (huof6829) [inline ioctl/newcmd/did/didgenerate_test.go:17]: move to group line 15. keep iotex-core in one single group

File: ioctl/newcmd/did/didgenerate_test.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	"github.com/iotexproject/iotex-core/ioctl/config"
+	"github.com/iotexproject/iotex-core/ioctl/util"
+	"github.com/iotexproject/iotex-core/test/mock/mock_ioctlclient"
+	"github.com/iotexproject/iotex-proto/golang/iotexapi/mock_iotexapi"
```
Comment (huof6829) [inline ioctl/newcmd/did/didgenerate_test.go:21]: the same

File: ioctl/newcmd/did/didgenerate_test.go [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+)
+
+func TestNewDidGenerate(t *testing.T) {
```
Comment (huof6829) [inline ioctl/newcmd/did/didgenerate_test.go:24]: TestNewDidGenerateCmd

File: ioctl/newcmd/did/didgenerate_test.go [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+	require := require.New(t)
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	client := mock_ioctlclient.NewMockClient(ctrl)
+	apiServiceClient := mock_iotexapi.NewMockAPIServiceClient(ctrl)
+	passwd := "123456"
+
+	ks := keystore.NewKeyStore(t.TempDir(), 2, 103)
```
Comment (huof6829) [inline ioctl/newcmd/did/didgenerate_test.go:32]: modify the last argument to 1.  keep the minimum consumption

File: ioctl/newcmd/did/didgenerate.go [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+			}
+
+			doc.Authentication = append(doc.Authentication, authentication)
+			msg, err := json.MarshalIndent(doc, "", "  ")
+			if err != nil {
+				return err
+			}
+
+			sum := sha256.Sum256(msg)
+			cmd.Printf("%s\n\nThe hex encoded SHA256 hash of the DID doc is:%s", string(msg), hex.EncodeToString(sum[:]))
```
Comment (huof6829) [inline ioctl/newcmd/did/didgenerate.go:97]: add  `\n` at the end

**Review by huof6829 — APPROVED** [review]

**Review by huangzhiran — APPROVED** [review]

**Pull request merged**